src/main/java/ch/naviqore/service/impl/PublicTransitServiceImpl.java

```diff
@@ -280,7 +280,7 @@ public Map<Stop, Connection> getIsolines(Stop source, LocalDateTime departureTim
 
     private Raptor getRaptor(LocalDate date) {
         // TODO: Not always create a new raptor, use mask on stop times based on active trips
-        return raptorCache.computeIfAbsent(date,
+        return raptorCache.computeIfAbsent(new HashSet<>(schedule.getActiveTrips(date)),
```

Comment:
I think this can be simplified by just adding the calendars that are active to the set.

schedule.getCalendars().values().stream().filter(calendar -> calendar.isServiceAvailable(date))

essentially the same thing, but I think the keys size will be smaller and there is less computation required.

Reply:
probably worth moving the key generator method to a separate method